syntax = "proto3";

package project;

import "portal_administrativo.proto";

message RelatorioDisciplina {
  Disciplina disciplina          = 1;
  Professor professor            = 2;
  repeated Aluno alunos          = 3;
}

message ResumoDisciplina {
  Disciplina disciplina          = 1;
  Professor professor            = 2;
  int32 totalAlunos              = 3;
}

message DisciplinaPessoa {
  // id da disciplina
  string disciplina = 1;
  // matricula do aluno ou siape do professor
  string idPessoa = 2;
}

service PortalMatricula {
  rpc AdicionaProfessor(DisciplinaPessoa) returns (Status) {}
  rpc RemoveProfessor(DisciplinaPessoa) returns (Status) {}
  rpc AdicionaAluno(DisciplinaPessoa) returns (Status) {}
  rpc RemoveAluno(DisciplinaPessoa) returns (Status) {}
  rpc DetalhaDisciplina(Identificador) returns (RelatorioDisciplina) {}
  rpc ObtemDisciplinasProfessor(Identificador) returns (stream RelatorioDisciplina) {}
  rpc ObtemDisciplinasAluno(Identificador) returns (stream ResumoDisciplina) {}
}